src/ophyd_async/sim/pattern_generator.py

```diff
@@ -81,74 +64,16 @@ def generate_interesting_pattern(x: float, y: float) -> float:
     return z
 
 
-class HdfStreamProvider:
-    def __init__(
-        self,
-        directory_info: DirectoryInfo,
-        full_file_name: Path,
-        datasets: List[DatasetConfig],
-    ) -> None:
-        self._last_emitted = 0
-        self._bundles: List[ComposeStreamResourceBundle] = self._compose_bundles(
-            directory_info, full_file_name, datasets
-        )
-
-    def _compose_bundles(
-        self,
-        directory_info: DirectoryInfo,
-        full_file_name: Path,
-        datasets: List[DatasetConfig],
-    ) -> List[StreamAsset]:
-        path = str(full_file_name.relative_to(directory_info.root))
-        root = str(directory_info.root)
-        bundler_composer = ComposeStreamResource()
-
-        bundles: List[ComposeStreamResourceBundle] = []
-
-        bundles = [
-            bundler_composer(
-                spec=SLICE_NAME,
-                root=root,
-                resource_path=path,
-                data_key=d.name.replace("/", "_"),
-                resource_kwargs={
-                    "path": d.path,
-                    "multiplier": d.multiplier,
-                    "timestamps": "/entry/instrument/NDAttributes/NDArrayTimeStamp",
-                },
-            )
-            for d in datasets
-        ]
-        return bundles
-
-    def stream_resources(self) -> Iterator[StreamResource]:
-        for bundle in self._bundles:
-            yield bundle.stream_resource_doc
-
-    def stream_data(self, indices_written: int) -> Iterator[StreamDatum]:
-        # Indices are relative to resource
-        if indices_written > self._last_emitted:
-            updated_stream_range = StreamRange(
-                start=self._last_emitted,
-                stop=indices_written,
-            )
-            self._last_emitted = indices_written
-            for bundle in self._bundles:
-                yield bundle.compose_stream_datum(indices=updated_stream_range)
-        return None
-
-    def close(self) -> None:
-        for bundle in self._bundles:
-            bundle.close()
-
-
 class PatternGenerator:
     def __init__(
         self,
         saturation_exposure_time: float = 1,
         detector_width: int = 320,
         detector_height: int = 240,
     ) -> None:
+        self.maxshape: tuple[Any, ...]
+        self.fillvalue: int = field(default_factory=int)
+        self.shape: List[int] = field(default_factory=[])
```

Comment:
These look like they were copied from a dataclass, where are their values?

Reply:
Is something like this what we want?

```suggestion
        self.maxshape: tuple[Any, ...] = (1,)
        self.fillvalue: int = 1
        self.shape: List[int] = [1, ...]
```

Reply:
I have pushed this change.